File: src/TgoeSrv/Member/Enums/MemberGroupCustomProperty.php
<?php
declare(strict_types = 1);
namespace TgoeSrv\Member\Enums;

enum MemberGroupCustomProperty : string
{
    case TRAINER = 'Trainer';
    case WEEKDAY = 'Tag';
    case TIME = 'Zeit';
    case LOCATION = 'Ort';
    case DOSB_SPORT = 'DOSB-Sportart';
    
    /**
     * Find enum value matching the key. Uses ignore case string comparison.
     *
     * @param string $key Key to search for.
     * @return DosbSport Enumeration value or NULL in case key does not exist.
     */
    public static function findByKey(string $key): ?MemberGroupCustomProperty
    {
        foreach( MemberGroupCustomProperty::cases() as $e) {
            if( $e->value == $key ) return $e;
        }
        
        return null;
    }
}

File: core/src/main/java/org/tgoe/core/member/enums/MemberGroupCustomProperty.java
package org.tgoe.core.member.enums;

public enum MemberGroupCustomProperty {
	TRAINER("Trainer"), WEEKDAY("Tag"), TIME("Zeit"), LOCATION("Ort"), DOSB_SPORT("DOSB-Sportart");

	private String key;

	private MemberGroupCustomProperty(String key) {
		this.key = key;
	}

	public String getKey() {
		return key;
	}

	/**
	 * Find enum value matching the key. Uses ignore case string comparison.
	 * 
	 * @param key Key to search for.
	 * @return Enumeration value or NULL in case key does not exist.
	 */
	public static MemberGroupCustomProperty findByKey(String key) {
		for( MemberGroupCustomProperty p : MemberGroupCustomProperty.values() ) {
			if( p.getKey().equalsIgnoreCase(key) ) return p;
		}
		
		return null;
	}
}
